mne/viz/_3d.py

```diff
@@ -1490,3 +1625,220 @@ def _get_view_to_display_matrix(scene):
+    show : bool
+        Show figure if True. Defaults to True.
+
+    Returns
+    -------
+    fig : instance of matplotlib figure
+        The figure containing 3-D locations of the dipoles.
+    """
+    import matplotlib.pyplot as plt
+    from mpl_toolkits.mplot3d import Axes3D
+    import scipy
+    from .. import Dipole
+    if has_nibabel():
+        import nibabel as nib
+        from nibabel.processing import resample_from_to
+    else:
+        raise ImportError('This function requires nibabel.')
+    if coord_frame not in ['head', 'mri']:
+        raise ValueError("coord_frame must be 'head' or 'mri'. "
+                         "Got %s." % coord_frame)
+    if idx == 'gof':
+        idx = np.argmax(dipole.gof)
+    elif idx == 'amplitude':
+        idx = np.argmax(dipole.amplitude)
+    elif not isinstance(idx, int):
+        raise ValueError("idx must be an int or one of ['gof', 'amplitude']. "
+                         "Got %s." % idx)
+
+    if not isinstance(dipole, Dipole):
+        dipole = Dipole(np.concatenate([d.times for d in dipole]),
+                        np.concatenate([d.pos for d in dipole]),
+                        np.concatenate([d.amplitude for d in dipole]),
+                        np.concatenate([d.ori for d in dipole]),
+                        np.concatenate([d.gof for d in dipole]))
+
+    subjects_dir = get_subjects_dir(subjects_dir=subjects_dir,
+                                    raise_error=True)
+    t1_fname = op.join(subjects_dir, subject, 'mri', 'T1.mgz')
+    t1 = nib.load(t1_fname)
+    vox2ras = t1.header.get_vox2ras_tkr()
+    ras2vox = scipy.linalg.inv(vox2ras)
+    trans = _get_trans(trans, fro='head', to='mri')[0]
+    zooms = t1.header.get_zooms()
+    if coord_frame == 'head':
+        affine_to = trans['trans'].copy()
+        affine_to[:3, 3] *= 1000  # to mm
+        aff = t1.affine.copy()
+
+        aff[:3, :3] /= zooms
+        affine_to = np.dot(affine_to, aff)
+        t1 = resample_from_to(t1, ([int(t1.shape[i] * zooms[i]) for i
+                                    in range(3)], affine_to))
+
+    data = t1.get_data()
+    dims = len(data)  # Symmetric size assumed.
+    dd = (dims / 2.)
+    if coord_frame == 'mri':
+        dd *= t1.header.get_zooms()[0]
+    fig = plt.figure()
+    ax = Axes3D(fig) if ax is None else ax
+    if coord_frame == 'head':
+        dipole_locs = np.array([apply_trans(ras2vox, loc * 1000.) * zooms
+                                for loc in dipole.pos])
+        ori = dipole.ori
+        scatter_points = dipole.pos * 1000.
+    else:
+        dipole_locs = np.array([apply_trans(trans['trans'], loc) for loc
+                                in dipole.pos])
+        ori = [apply_trans(trans['trans'], o) for o in dipole.ori]
+        # Position in meters -> voxels.
+        dipole_locs = np.array([apply_trans(ras2vox, loc * 1000.) for loc
+                                in dipole_locs])
+        scatter_points = apply_trans(vox2ras, dipole_locs)
```

Comment:
Can this be refactored earlier (so there is noly one `if coord_frame == 'head'`)? For example you can make a `dip_trans` above that is identity for head coords and the `head_mri_t` for MRI coords, etc. This is how it's done in `plot_trans` and it really makes the code cleaner and easier to follow.